### PR TITLE
Clone tags within drone

### PR DESCRIPTION
**Pull request by appleboy**
Clone tags within drone for proper version generation

### CODE REVIEW

**Comment by appleboy** [comment]
@ipedrazas You need to resign drone sig file.

**Comment by ipedrazas** [comment]
Not sure I understand this change...

File: .drone.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ workspace:
   base: /go
 
 pipeline:
+  clone:
+    image: plugins/git
+    tags: true
```
Comment (appleboy) [inline .drone.yml]: https://github.com/appleboy/drone-scp/blob/master/Makefile#L22-L26

Default clone behavior does not include all tags so you need to set `TAG` in environment section.

You can try it out by yourself.

Comment (ipedrazas) [inline .drone.yml]: But surely that's a change you want on your drone file, not in this one... The tag that you just removed is a docker image tag, not a git tag

Comment (appleboy) [inline .drone.yml]: `DRONE_TAG` default value is `${DRONE_BRANCH}-${DRONE_COMMIT_SHA:0:7}` if you set clone `tags` as `true`.

Comment (appleboy) [inline .drone.yml]: So I don't know why you need to set new environment `TAG`, you can just use `${DRONE_TAG}` which drone already supported.

Comment (ipedrazas) [inline .drone.yml]: I'm not sure, if I use `tags` as `true` this is what I get:

```
Successfully built 5ee8ab55d52a
+ docker tag image quay.io/ipedrazas/drone-helm:latest
+ docker tag image quay.io/ipedrazas/drone-helm:${DRONE_TAG}
Error parsing reference: "quay.io/ipedrazas/drone-helm:" is not a valid repository/tag
```
as you can see `${DRONE_TAG}` is empty and yes, the clone tags block is added:

```
+ git init
Initialized empty Git repository in /go/src/github.com/ipedrazas/drone-helm/.git/
+ git remote add origin https://github.com/ipedrazas/drone-helm.git
+ git fetch --tags origin +refs/heads/clonetags:
From https://github.com/ipedrazas/drone-helm
 * branch            clonetags  -> FETCH_HEAD
 * [new branch]      clonetags  -> origin/clonetags
+ git reset --hard -q 5b91a1b48905b43d05311762635899d8a4b83043
+ git submodule update --init --recursive
```

Comment (appleboy) [inline .drone.yml]: This is example of gorush repo: https://github.com/appleboy/gorush/blob/master/.drone.yml#L44-L50

Drone Log: http://drone.wu-boy.com/appleboy/gorush/267/1

You can see git clone all tag if set `tags` as `true`.

Comment (ipedrazas) [inline .drone.yml]: I think there's some confusion somewhere. You are right, git clone all tags if set as true. However, I'm not using git tags. What I have is a `${DRONE_TAG}` envvar that contains my branch and my git sha.

Correct me if I'm wrong, but git clone doesn't create that envvar. [The link](https://github.com/appleboy/drone-scp/blob/master/Makefile#L22-L26) that you provided uses that envvar, but when [I tried to use it](https://github.com/ipedrazas/drone-helm/commit/5b91a1b48905b43d05311762635899d8a4b83043#diff-3216dfff0ed3e301453e6799e8c367e2R27), it was empty.

If you look at [the definition](https://github.com/drone/drone/blob/b7a0482c68fa5eac54e0e188bfed205efd685af8/agent/agent.go#L308) of the ${DRONE_TAG} envvar:

```
if w.Build.Event == model.EventTag {
	envs["DRONE_TAG"] = strings.TrimPrefix(w.Build.Ref, "refs/tags/")
}
```
`w.Build.Ref` is formed by `refs/heads/` and the branch name, like this: `refs/heads/mybranch`

So, I see where you're trying to go, but I do not use git tags, and the docker image tag that I want to create is formed by these two strings:

`TAG=${DRONE_BRANCH}-${DRONE_COMMIT_SHA:0:7}`

Comment (appleboy) [inline .drone.yml]: @ipedrazas My apologies. You can refer to Makefile of Gorush project. https://github.com/appleboy/gorush/blob/master/Makefile#L22-L26

Your config is correct if you don't have check `DRONE_TAG` variable on Makefile or something else.

Comment (appleboy) [inline .drone.yml]: Sorry for wasting your time.

**Pull request closed without merging**